### PR TITLE
Add support for meta description

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/wpilibsuite/sphinxext-opengraph/issues/65.

Re: https://github.com/wpilibsuite/sphinxext-opengraph/issues/65#issuecomment-1285497461.

When `ogp_enable_meta_description=True`, only add a generated `<meta name="description" content="...">` tag if the page doesn't already have a manual one.

For example, a manual override may have been added using this RST:

```rst
.. meta::
   :description: My manual description
```

---

Also fix a bug where it was adding multiple meta descriptions for subpages.

TODO:

* [x] Rename config option to `ogp_enable_meta_description`
* [x] Document new option
* [x] Address https://github.com/wpilibsuite/sphinxext-opengraph/issues/65#issuecomment-1286542474

### CODE REVIEW

**Comment by TheTripleV** [comment]
Looks good so far. Please see the comments on #65 and #71

**Comment by hugovk** [comment]
Rebased to resolve conflicts, and renamed option to `ogp_enable_meta_description`. Still need to address https://github.com/wpilibsuite/sphinxext-opengraph/issues/65#issuecomment-1286542474 and document.

**Comment by hugovk** [comment]
Updated for https://github.com/wpilibsuite/sphinxext-opengraph/issues/65#issuecomment-1286542474.

Still TODO: add documentation.

**Comment by hugovk** [comment]
Docs added, ready for review!

File: sphinxext/opengraph/__init__.py [hunk 1]
```diff
@@ -186,6 +195,7 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     app.add_config_value("ogp_type", "website", "html")
     app.add_config_value("ogp_site_name", None, "html")
     app.add_config_value("ogp_custom_meta_tags", [], "html")
+    app.add_config_value("ogp_enable_meta_description", False, "html")
```
Comment (TheTripleV) [inline sphinxext/opengraph/__init__.py:198]: I think enabling this by default would be better.

**Review by TheTripleV — CHANGES_REQUESTED** [review]

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -4,8 +4,6 @@ on:
     branches:
       - main
   push:
-    branches:
-      - main
```
[the remark this answers is not in the document]
Comment (hugovk) [reply .github/workflows/workflow.yml:8]: I can put it back but it's a big impediment to contribution.

With this restriction, it means I can only test my PRs on CI if I:

* temporarily comment it out to begin, then uncomment it later, or
* don't create feature branches and develop on `main`, not a best practice and limiting to being able to work on one thing at a time, or
* don't test my work on CI at all before creating a PR, meaning I've not tested it on all required Python, OS and Sphinx versions

CIs are very powerful and they really help improve quality, I believe we should be encouraging contributors to test via CI first.

---

Would something like this help?

```yml
    # We want to run on external PRs, but not on our own internal PRs as they'll be run
    # by the push to the branch.
    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```

https://github.com/has2k1/plotnine/blob/1da5d2b739e904f2c1ce97d2ba4e3a84129d0332/.github/workflows/testing.yml#L10-L12

File: sphinxext/opengraph/__init__.py (hunk 1 above)
Comment (hugovk) [reply sphinxext/opengraph/__init__.py]: Updated!

File: .github/workflows/workflow.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheTripleV) [reply .github/workflows/workflow.yml:8]: Sure, that seems fine to me.

[the remark this answers is not in the document]
Comment (hugovk) [reply .github/workflows/workflow.yml:8]: Thanks, updated!

**Review by TheTripleV — APPROVED** [review]

**Pull request merged**